src/codecs/processor-worker/index.ts

```diff
@@ -31,13 +33,36 @@ async function rotate(
 async function resize(
   data: ImageData, opts: import('../resize/processor-meta').WorkerResizeOptions,
 ): Promise<ImageData> {
+  if (isHqx(opts)) {
+    const { hqx } = await import(
+      /* webpackChunkName: "process-hqx" */
+      '../hqx/processor');
+
+    const widthRatio = opts.width / data.width;
+    const heightRatio = opts.height / data.height;
+    const ratio = Math.max(widthRatio, heightRatio);
+    const factor = clamp(Math.floor(ratio), { min: 2, max: 4 }) as 2|3|4;
+    if (ratio < 2) {
+      return data;
+    }
```

Comment:
Feels like we should only bail if ratio <= 1. This could be moved up a line. Also we're allowed one-liners in this project, so `if (ratio <=1) return data` ftw.